qt5ct: 0.24 -> 0.30

romildo opened the pull request:
###### Motivation for this change

There are new releases.

[Changelog:](https://www.linux-apps.com/p/1131604/)

**0.30**

fixed crash on nonexistent style in qt5ct.conf
fixed issue with KDE color themes
updated documentation
updated Bulgarian translation
updated French translation

**0.29**

fixed libqt5ct-style.so installation path
updated Greek translation

**0.28**

added Qt 5.8 support
added qt5ct proxy-style plugin
added Serbian translation
updated Bulgarian translation
updated Italian translation
updated Russian translation
updated Dutch (Netherlands) translation
updated Czech translation
updated German translation
updated Chinese (Taiwan) translation

**0.27**

added D-Bus tray support
added feature to disable debug output
updated documentation

**0.26**

fixed stellarium compatibility
updated French translation
updated Greek translation

**0.25**

added environment variables verification
added Frensh translation
updated Slovak translation
updated Russian translation
updated Czech translation
updated German translation
updated Dutch (Netherlands) translation
updated Polish translation
updated Chinese (Taiwan) translation

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

romildo commented:
Currently it does not work on my NixOS system, though.

```
$ qt5ct &
Configuration path: /home/romildo/.config/qt5ct/
Shared QSS path: /nix/store/8zffs05x029qw4i8qxg9wjjqfc5rsqfb-qt5ct-0.30/share/qt5ct/qss/

$ ls -l /nix/store/8zffs05x029qw4i8qxg9wjjqfc5rsqfb-qt5ct-0.30/share/
total 4
dr-xr-xr-x 2 root root 4096 Dec 31  1969 applications

$ ls -l /nix/store/8zffs05x029qw4i8qxg9wjjqfc5rsqfb-qt5ct-0.30/lib/qt5/plugins/platformthemes/styles
total 36
-r-xr-xr-x 1 root root 34840 Dec 31  1969 libqt5ct-style.so
```
![qt5ct-error](https://cloud.githubusercontent.com/assets/1217934/23461211/cabab940-fe67-11e6-98db-f7812553b4e6.png)

I am not expert on Qt. Maybe someone knows how to deal with these issues and want to help.

romildo commented:
The README at [qt5ct](https://sourceforge.net/p/qt5ct/code/HEAD/tree/tags/qt5ct-0.30/) gives some hints on how to fix these issues:
- `export QT_QPA_PLATFORMTHEME=qt5ct`
- `qmake PLUGINDIR=<custom path>`
With the latest patch it works for me.

Ralith commented on pkgs/tools/misc/qt5ct/default.nix:
```diff
@@ -2,22 +2,22 @@
 
   preFixup = ''
-    wrapQtProgram $out/bin/qt5ct
+    wrapQtProgram $out/bin/qt5ct --set QT_QPA_PLATFORMTHEME qt5ct
```
This environment variable should be set in your user environment (e.g. through .xprofile) so that it affects all Qt applications. Setting it in the wrapper seems liable to just confuse people when it has no external effect.

romildo replied:
Probably it should be set in a module then.

Ralith commented:
I don't seem to be able to run this on my 16.09 system (`This application failed to start because it could not find or load the Qt platform plugin "xcb" in "".`) but I suspect that's due to unrelated impurities in the Qt package pertaining to the update to 5.8, so it shouldn't be a blocker.

romildo committed:
qt5ct: 0.24 -> 0.30

romildo committed:
qt5ct: add a nixos module to enable qt5ct

In order to use qt5ct (Qt5 Configuration Tool) to configure Qt5
settings (theme, font, icons, etc.) under DE/WM without Qt
integration, the environment variable QT_QPA_PLATFORMTHEME should be
set to "qt5ct".

It can be done automatically by this module, or by setting the
variable explicitly in the user or in the system configuration.

romildo commented:
In this latest version of the PR the environment variable `QT_QPA_PLATFORMTHEME` is not set in the wrapper anymore. A module is added for that. In `configuration.nix` it is enough to set
```
  programs.qt5ct.enable = true;
```
to enable the Qt5 Configuration Tool.

I have tested on a virtual machine and it works for me.

Ralith commented on nixos/modules/programs/qt5ct.nix:
```diff
@@ -0,0 +1,31 @@
+
+  ###### implementation
+  config = mkIf config.programs.qt5ct.enable {
+    environment.variables.QT_QPA_PLATFORMTHEME = [ "qt5ct" ];
```
I think this should be a string, not a list of strings.

romildo replied:
Fixed now.

Ralith reviewed (changes requested):
One last quibble, but otherwise I'm satisfied with this approach.

Ralith reviewed (approved):
Thanks for the update!

romildo commented:
Maybe somebody could merge this PR. It is becoming too old.
